numpy/lib/histograms.py

```diff
@@ -777,16 +776,39 @@ def histogram(a, bins=10, range=None, normed=False, weights=None,
 
     # density overrides the normed keyword
     if density is not None:
+        if normed is not None:
+            # 2018-06-13, numpy 1.16.0 (this was not noisily deprecated in 1.6)
+            warnings.warn(
+                    "The normed argument is ignored when density is provided. "
+                    "In future passing both will result in an error."
+                    DeprecationWarning, stacklevel=2)
         normed = False
 
     if density:
         db = np.array(np.diff(bin_edges), float)
         return n/db/n.sum(), bin_edges
     elif normed:
-        # deprecated, buggy behavior. Remove for NumPy 2.0.0
+        # 2018-06-13, numpy 1.16.0 (this was not noisily deprecated in 1.6)
+        warnings.warn(
+                "Passing `normed=True` on non-uniform bins has always been "
+                "broken, and computes neither the probability density "
+                "function nor the probability mass function. "
+                "The result is only correct if the bins are uniform, when "
+                "density=True will produce the same result anyway. "
+                "The argument will be removed in a future version of "
+                "numpy.",
+                DeprecationWarning, stacklevel=2)
```

Comment:
Maybe even VisibleDeprecationWarning, since I am not sure how good the reporting is by now for the warning, and it can be a bug?

I think I would prefer the text to come to the point faster, maybe something in the direction of:
```
passing `normed=True` to histogram has been deprecated, instead use `density=True`.\n
`normed` produces incorrect results if the bins are not uniform and will thus be
removed in favor of `density` which always produces the correct result.
```

Reply:
Feel free to adjust the wording in the PR - that suggestion looks reasonable